amaranth/vendor/lattice_nexus.py

```diff
@@ -0,0 +1,675 @@
+                {% endif %}
+            {% endfor %}
+            {{get_override("add_preferences")|default("# (add_preferences placeholder)")}}
+        """,
+        "{{name}}.sdc": r"""
+            {% for net_signal, port_signal, frequency in platform.iter_clock_constraints() -%}
+                {% if port_signal is not none -%}
+                    create_clock -name {{port_signal.name|tcl_escape}} -period {{1000000000/frequency}} [get_ports {{port_signal.name|tcl_escape}}]
+                {% else -%}
+                    create_clock -name {{net_signal.name|tcl_escape}} -period {{1000000000/frequency}} [get_nets {{net_signal|hierarchy("/")|tcl_escape}}]
+                {% endif %}
+            {% endfor %}
+            {{get_override("add_constraints")|default("# (add_constraints placeholder)")}}
+        """,
+    }
+    _radiant_command_templates = [
+        # These don't have any usable command-line option overrides.
+        r"""
+        {{invoke_tool("radiantc")}}
+            {{name}}.tcl
+        """,
+        r"""
+        {{invoke_tool("programmer")}}
+            -oft -bit
+            -if {{name}}_impl/{{name}}_impl.bit -of {{name}}.bit
+        """,
+        r"""
+        {{invoke_tool("programmer")}}
+            -oft -xcfsingle -revd -op "Fast Program"
+            -if {{name}}_impl/{{name}}_impl.bit -of {{name}}.xcf
+        """,
+    ]
+
+    # Common logic
+
+    def __init__(self, *, toolchain="Oxide"):
+        super().__init__()
+
+        assert toolchain in ("Oxide", "Radiant")
+        self.toolchain = toolchain
+
+    @property
+    def required_tools(self):
+        if self.toolchain == "Oxide":
+            return self._oxide_required_tools
+        if self.toolchain == "Radiant":
+            return self._radiant_required_tools
+        assert False
+
+    @property
+    def file_templates(self):
+        if self.toolchain == "Oxide":
+            return self._oxide_file_templates
+        if self.toolchain == "Radiant":
+            return self._radiant_file_templates
+        assert False
+
+    @property
+    def command_templates(self):
+        if self.toolchain == "Oxide":
+            return self._oxide_command_templates
+        if self.toolchain == "Radiant":
+            return self._radiant_command_templates
+        assert False
+
+    @property
+    def default_clk_constraint(self):
+        if self.default_clk == "OSC":
+            return Clock(150e6 / self.osc_div) # lscc/radiant/2022.1/docs/webhelp/eng/index.htm#page/Reference%20Guides/FPGA%20Libraries%20Reference%20Guide/Alphanumeric_Primitives_List.11.155.htm        
+        # For HFCLKOUT (created clock): ADJUSTED_PERIOD=6666.66667ps * HF_CLK_DIV
+        return super().default_clk_constraint
+
+    def create_missing_domain(self, name):
+        # Lattice ECP5 devices have two global set/reset signals: PUR, which is driven at startup
```

Comment:
Should this be changed to mention a different family?

Reply:
yes